include/pros/link.h

```diff
@@ -0,0 +1,252 @@
+ */
+
+#ifndef _PROS_LINK_H_
+#define _PROS_LINK_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+namespace pros {
+#endif
+
+typedef enum link_type_e {
+    E_LINK_RECIEVER = 0,
+    E_LINK_TRANSMITTER
+} link_type_e_t;
+
+#ifdef __cplusplus
+namespace c {
+#endif
+
+/**
+ * Initializes a link on a radio port, with an indicated type.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as a radio.
+ * ENXIO - The sensor is still calibrating, or no link is connected via the radio.
+ *
+ * \param port 
+ *      The port of the radio for the intended link.
+ * \param link_id
+ *      Unique link ID in the form of a string, needs to be different from other links in
+ *      the area.
+ * \param type
+ *      Indicates whether the radio link on the brain is a transmitter or reciever,
+ *      with the transmitter having double the transmitting bandwidth as the recieving
+ *      end (1040 bytes/s vs 520 bytes/s).
+ *
+ * \return PROS_ERR if initialization fails, 1 if the initialization succeeds.
+ */
+uint32_t link_init(uint8_t port, char* link_id, link_type_e_t type);
+
+/**
+ * Initializes a link on a radio port, with an indicated type and the ability for
+ * vexlink to override the controller radio.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as a radio.
+ * ENXIO - The sensor is still calibrating, or no link is connected via the radio.
+ *
+ * \param port 
+ *      The port of the radio for the intended link.
+ * \param link_id
+ *      Unique link ID in the form of a string, needs to be different from other links in
+ *      the area.
+ * \param type
+ *      Indicates whether the radio link on the brain is a transmitter or reciever,
+ *      with the transmitter having double the transmitting bandwidth as the recieving
+ *      end (1040 bytes/s vs 520 bytes/s).
+ *
+ * \return PROS_ERR if initialization fails, 1 if the initialization succeeds.
+ */
+uint32_t link_init_override(uint8_t port, char* link_id, link_type_e_t type);
+
+/**
+ * Checks if a radio link on a port is active or not.
+ * 
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as a radio.
+ * ENXIO - The sensor is still calibrating, or no link is connected via the radio.
+ * 
+ * \param port 
+ *      The port of the radio for the intended link.
+ *
+ * \return If a radio is connected to a port and it's connected to a link.
+ */
+bool link_connected(uint8_t port);
+
+/**
+ * Returns the bytes of data number of without protocol available to be read 
```

Comment:
You probably don't want to document the protocol, but you'll probably need to document the max size that it takes up. I'm not sure "your message needs to be sufficiently smaller then link_raw_receivable_size" will be sufficiently helpful, especially since this is a convenience layer in itself.